ambry-cloud/src/main/java/com/github/ambry/cloud/CloudDestination.java

```diff
@@ -137,4 +139,13 @@ boolean retrieveTokens(String partitionPath, String tokenFileName, OutputStream
    * Halt any compactions in progress.
    */
   void stopCompaction();
+
+  /**
+   * Deprecate the specified {@link Container}s for specified partitions in cloud.
+   * @param deprecatedContainers {@link Set} of deprecated {@link Container}s.
+   * @param partitionIds {@link Collection} of cloud partition ids from where the container needs to be deprecated.
+   * @throws {@link CloudStorageException} if updating fails.
+   */
+  void deprecateContainers(Set<Container> deprecatedContainers, Collection<String> partitionIds)
```

Comment:
Minor: Let's use `Collection` instead of `Set` for `accounts/updateContainers` to make it more general.

Reply:
Also, I wonder in which case the `partitionIds` don't equal to `allPartitionIds`? I am thinking if we can remove the partitionIds because container-to-partitions mapping is not clear to me. 
(Maybe it's not easy but I am trying to say the partitions might be passed into cloudDestination class, not in this method)